### PR TITLE
Begin 1.19.2 protocol.json updates

### PR DESCRIPTION
**Pull request by jtsiskin**
Note the 4 new packet types (git doesn't do a good job of displaying this, we could probably store these mappings as a list eventually) which are not yet implemented - 

```
chat_suggestions
hide_message
message_header
message_acknowledgement
```

### CODE REVIEW

File: data/pc/1.19.2/protocol.json [hunk 1]
```diff
@@ -5026,6 +5110,59 @@
             {
               "name": "signedPreview",
               "type": "bool"
+            },
+            {
+              "name": "previousMessages",
```
Comment (extremeheat) [inline data/pc/1.19.2/protocol.json:5115]: we can make this a shared type since it's the same fields in player_chat

File: data/pc/1.19.2/protocol.json [hunk 2]
```diff
@@ -3195,13 +3208,77 @@
...
-                "buffer",
+                "array",
                 {
-                  "countType": "varint"
+                  "countType": "varint",
+                  "type": [
+                    "container",
+                    [
+                      {
+                        "name": "previousMessageSender",
```
Comment (extremeheat) [inline data/pc/1.19.2/protocol.json:3220]: this can just be "messageSender" and "messageSignature" as previous is implied

**Review by extremeheat — COMMENTED** [review]
looks good, just left some comments

File: data/pc/common/features.json [hunk 3]
```diff
@@ -664,6 +664,14 @@
     "description": "allows for usage of `signature` instead of verifyToken in serverbound packet_encryption_begin",
     "versions": ["1.19_major", "latest"]
   },
+  {
+    "name": "signatureV2",
```
Comment (extremeheat) [inline data/pc/common/features.json:668]: try "chatSignatureV2" to be less ambiguous

File: data/pc/1.19.2/protocol.json (hunk 2 above)
Comment (jtsiskin) [reply data/pc/1.19.2/protocol.json]: done

File: data/pc/common/features.json (hunk 3 above)
Comment (jtsiskin) [reply data/pc/common/features.json]: changed to `profileKeySignatureV2` to specify this is the format used when signing key during initial handshake

**Comment by rom1504** [comment]
Did you implement all packets changes ?
Can you post a short list of the changes and how you got the information?

**Comment by rom1504** [comment]
Are all changes implemented now ?

**Comment by jtsiskin** [comment]
> Are all changes implemented now ?

Yes

**Comment by SpencerBurgess** [comment]
Don't' forget to update the README to reflect the new support version

**Comment by rom1504** [comment]
packets that changed:
* packet_multi_block_change
* packet_explosion
* packet_player_chat
* packet_respawn
* packet_system_chat
* packet_server_data
* packet_chat_command
* packet_chat_message
* entityMetadataItem
* packet_chat_suggestions
* packet_hide_message
* packet_message_header
* packet_message_acknowledgement

**Comment by rom1504** [comment]
I'm surprised so much changed

also sad that wiki.vf seems to have stopped using pre release page so not possible to check there

**Comment by rom1504** [comment]
still got https://wiki.vg/Protocol history

**Comment by IceTank** [comment]
`packet_explosion` should have f32 as x y z coordinates. https://wiki.vg/Protocol#Explosion Also nitpick wiki.vg calls Prismarines `radius` variable `strength` not radius.
Don't know what changed about packet_multi_block_change. Looks like it is the same on wiki.vg for 1.19 and 1.19.2?

**Comment by rom1504** [comment]
https://wiki.vg/Protocol_version_numbers says
* https://wiki.vg/index.php?title=Protocol&oldid=17753 is 1.19
* https://wiki.vg/Protocol is 1.19.2
it may help checking diffs

along with https://wiki.vg/index.php?title=Protocol&action=history

**Comment by rom1504** [comment]
also https://github.com/extremeheat/extracted_minecraft_data/pull/1/files#diff-d3466db9ef694cae789954e4cce3ea38df5e39ce0c8c9be605be21e4cecb963b was used as a reference

File: data/pc/1.19.2/protocol.json [hunk 4]
```diff
@@ -2132,7 +2139,7 @@
               ]
             },
             {
-              "name": "notTrustEdges",
+              "name": "suppressLightUpdates",
```
Comment (rom1504) [inline data/pc/1.19.2/protocol.json:2174]: why was this changed? was a field deleted and another one introduced in the same place ?

**Comment by rom1504** [comment]
will come back to this if we see it doesn't work
